client/src/main/java/se/fortnox/reactivewizard/client/JsonArrayDeserializer.java

```diff
@@ -0,0 +1,69 @@
+package se.fortnox.reactivewizard.client;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.async.ByteArrayFeeder;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.databind.util.TokenBuffer;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonArrayDeserializer {
+    private final ObjectReader reader;
+    private final JsonParser parser;
+    private final ByteArrayFeeder inputFeeder;
+
+    public JsonArrayDeserializer(ObjectMapper objectMapper, Method method) {
+        Type type = ReflectionUtil.getTypeOfObservable(method);
+
+        JavaType javaType = TypeFactory.defaultInstance().constructType(type);
+        reader   = objectMapper.readerFor(javaType);
+        try {
+            parser = objectMapper.getFactory().setCodec(objectMapper).createNonBlockingByteArrayParser();
+            inputFeeder = (ByteArrayFeeder) parser.getNonBlockingInputFeeder();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Publisher<?> process(byte[] buf) {
```

Comment:
will it work with nested objects?

Reply:
I belive so, but will add a test to prove it :-)

Reply:
It would also be nice to have tests for compatibility with the "old-style" streaming method. (Sequential objects without delimiters)

Reply:
Like this?
```
{"name":"A"}{"name":"B"}
```
Not sure we want to support that in the client. Wouldn't it be strange if that did not give an error for bad json?

Reply:
Perphaps switch behavior on a "legacy" marker annotation? I would like to be able to call currently existing streaming resources without much hassle.

Reply:
The old @Stream annotated methods is an Observable<SMTH> and that will still work i guess but no streaming when calling. 
The @Stream annotation was only intended for the server side.

If you want to make use of the new stuff before the old @Stream annotated endpoint is migrated to Flux<SMTH> then you could easily create your own interface in your own codebase mimicing the one from the maven dependency and just change the return type in the local interface.

Reply:
@rassmate Sure, that's exactly what I would like to do, but I don't know of a way to declare such an interface if the new `Flux`-resources are unable to parse the current `@Stream`-responses.

Reply:
No, my bad, thought it might work out of the box but it doesn't due to everything you written here. Just a slow learner...

Reply:
Well, current implementations with `@Stream` has the annotation on the impl and not on the api, so I don't see how I can detect this on the client side.

I could change the server side of `@Stream` to emit valid json, but that would be a "breaking" change. Most usage of `@Stream` is for text/csv anyway, I have never used it for json, so that might be a way to go? @TommyPersson 

After this PR is merged, you could just change or add a new endpoint with a Flux return type to use the new client, would that work for you? It can co-exist with existing endpoints in the same class.

Reply:
There is an internal service that streams json and it's what mostly motivated the other PR. But I can accept waiting for that service to be updated with new proper streaming behavior.